actions/database-actions/write-document.js

```diff
@@ -62,8 +63,13 @@ function insertOrUpdate(cloudantDb, overwrite, doc) {
                                reject(err);
                             });
                     } else {
-                        console.error('error', error);
-                        reject(error);
+                        if(error.statusCode === 404) {
+                            // If document not found, insert it
+                            return insert(cloudantDb, doc);
```

Comment:
the promise that contains the cloudant "get" is never resolved for this path

Reply:
good catch, looks like insert creates it own promise.

Reply:
let me update that

Reply:
updated